lua/telescope/builtin/git.lua

```diff
@@ -98,26 +92,42 @@ git.branches = function(opts)
     upstream = 0,
     committerdate = 0,
   }
-  local register_entry = function(entry, trim_refname_prefix)
-    entry.name = string.sub(entry.refname, string.len(trim_refname_prefix)+1)
+  local unescape_single_quote = function(v)
+      return string.gsub(v, "\\([\\'])", "%1")
+  end
+  local parse_line = function(line)
+    local fields = vim.split(string.sub(line, 2, -2), "''", true)
+    local entry = {
+      head = fields[1],
+      refname = unescape_single_quote(fields[2]),
+      authorname = unescape_single_quote(fields[3]),
+      upstream = unescape_single_quote(fields[4]),
+      committerdate = fields[5],
+    }
+    local prefix
```

Comment:
is this `prefix` only used for line 120? If so, I think it'd be better to use number directly so we don't need to calculate the length using `string.len`.